datafusion/common/src/dfschema.rs

```diff
@@ -33,17 +33,36 @@ use crate::{
 
 use arrow::compute::can_cast_types;
 use arrow::datatypes::{DataType, Field, FieldRef, Fields, Schema, SchemaRef};
+use sqlparser::ast::Table;
 
 /// A reference-counted reference to a `DFSchema`.
 pub type DFSchemaRef = Arc<DFSchema>;
 
 /// DFSchema wraps an Arrow schema and adds relation names
+///
+/// # Example
+/// ```
+/// Creating a DF schema from an arrow schema
+/// ```
+///
+/// ```
+/// Converting from DF schema to arrow schema
+/// ```
+///
+/// ```
+/// Iterating over qualified fields
+/// ```
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct DFSchema {
-    /// Fields
-    fields: Vec<DFField>,
-    /// Additional metadata in form of key value pairs
-    metadata: HashMap<String, String>,
+    /// Inner arrow schema
+    inner: SchemaRef,
+
+    /// Optional qualifiers for each column in this schema. In the same order as
```

Comment:
This is the key design --remove `DFField` and simply wrap an inner arrow Schema ref.